src/toxcore_tests/packet_kind_tests.rs

```diff
@@ -64,3 +66,15 @@ fn packet_kind_from_bytes_test() {
     with_bytes(vec![3]);  // incorrect
     with_bytes(vec![4]);
 }
+
+// PacketKind::parse_bytes
+
+#[test]
+fn packet_kind_parse_bytes_rest_test() {
+    fn with_bytes(bytes: Vec<u8>) {
+        if let Ok(Parsed(_, rest)) = PacketKind::parse_bytes(&bytes) {
+            assert_eq!(&bytes[1..], rest);
+        }
```

Comment:
`if` branch should return `TestResult::passed()`, and there should be added `else` branch that would return `TestResult::discard()`; at this point though `match` would be better.

e.g. https://github.com/zetok/tox/blob/771ef573e620ad672f53d1c801e16bdc3f506760/src/toxencryptsave_tests/encryptsave_tests.rs#L50,L63